python/dgl/nn/pytorch/network_emb.py

```diff
@@ -0,0 +1,220 @@
+"""Network Embedding NN modules"""
+# pylint: disable= invalid-name
+import torch
+from torch import nn
+from torch.nn import init
+from torch.utils.data import DataLoader
+import torch.nn.functional as F
+import numpy as np
+import tqdm
+from ...sampling import random_walk
+
+
+
+class Metapath2vec(nn.Module):
+    """
+    Metapath2vec, a scalable representation learning model for heterogeneous networks
+    from KDD’17, August <https://dl.acm.org/doi/pdf/10.1145/3097983.3098036>
+    Parameters
+    ----------
+    g :DGLHeteroGraph
+        Graph data for generating embedding
+    emb_dimension :int
+        Size of each embedding vector
+    metapath :List[str]
+        The metapath is described as a list of strings for relation edges between different
+        types of nodes. It is a  sequence of edges which denotes transition  for different
+        types of nodes at each step. Note that the node type at the start and end of the
+        metapath are commonly the same.
+    window_size :int
+        The context size which is considered for positive samples.
```

Comment:
Perhaps you need to explain what are positive and negative samples at the beginning and how they affect the training.